engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala

```diff
@@ -1,61 +1,79 @@
 package cromwell.jobstore
 
-import cromwell.core.{JobOutputs, WorkflowId}
+import cromwell.core.ExecutionIndex._
+import cromwell.core.WorkflowId
+import cromwell.core.simpleton.WdlValueSimpleton._
+import cromwell.core.simpleton.{WdlValueBuilder, WdlValueSimpleton}
 import cromwell.database.sql.JobStoreSqlDatabase
-import cromwell.database.sql.tables.JobStoreEntry
+import cromwell.database.sql.tables.{JobStoreEntry, JobStoreResultSimpletonEntry}
+import wdl4s.TaskOutput
+import wdl4s.types._
+import wdl4s.values.WdlPrimitive
 
 import scala.concurrent.{ExecutionContext, Future}
-import cromwell.core.ExecutionIndex._
-import spray.json._
 
 class SqlJobStore(sqlDatabase: JobStoreSqlDatabase) extends JobStore {
   override def writeToDatabase(jobCompletions: Map[JobStoreKey, JobResult], workflowCompletions: List[WorkflowId])(implicit ec: ExecutionContext): Future[Unit] = {
     for {
-      _ <- sqlDatabase.addJobStoreEntries(jobCompletions map toJobStoreEntry)
+      _ <- sqlDatabase.addJobStoreEntries(jobCompletions map toDatabase)
       _ <- sqlDatabase.removeJobResultsForWorkflows(workflowCompletions map {_.toString})
     } yield ()
   }
 
-  private def toJobStoreEntry(jobCompletion: (JobStoreKey, JobResult)): JobStoreEntry = {
-    import JobResultJsonFormatter._
+  private def toDatabase(jobCompletion: (JobStoreKey, JobResult)): (JobStoreEntry, Iterable[JobStoreResultSimpletonEntry]) = {
     jobCompletion match {
-      case (key, JobResultSuccess(returnCode, jobResult)) => JobStoreEntry(
-        key.workflowId.toString,
-        key.callFqn,
-        key.index.fromIndex,
-        key.attempt,
-        jobSuccessful = true,
-        returnCode,
-        Some(jobResult.toJson.toString),
-        None,
-        None)
-      case (key, JobResultFailure(returnCode, throwable, retryable)) => JobStoreEntry(
-        key.workflowId.toString,
-        key.callFqn,
-        key.index.fromIndex,
-        key.attempt,
-        jobSuccessful = false,
-        returnCode,
-        None,
-        Some(throwable.getMessage),
-        Some(retryable))
+      case (key, JobResultSuccess(returnCode, jobOutputs)) =>
+        val entry = JobStoreEntry(
+          key.workflowId.toString,
+          key.callFqn,
+          key.index.fromIndex,
+          key.attempt,
+          jobSuccessful = true,
+          returnCode,
+          None,
+          None)
+        val simpletons = jobOutputs.mapValues(_.wdlValue).simplify.map {
+          s => JobStoreResultSimpletonEntry(s.simpletonKey, s.simpletonValue.valueString, s.simpletonValue.wdlType.toWdlString, -1) }
```

Comment:
ToL: Creating the database row here with a `-1` for the foreign key is very un-scala-esque. Almost like passing in a `null` to be filled in later. For primary keys, we've used `Option[_]` to represent the auto-incremented keys populated by the database. For most fk's though I've jumped through hoops to pass in a function-that-takes-the-fk-and-returns-the-new-rows. Using `-1` is much, much simpler though than even using an `Option` for the fk, and definitely simpler than passing in a function parameter. I'm torn on what's most "correct" though.